pkg/networkserver/networkserver.go

```diff
@@ -77,11 +78,16 @@ type newDevAddrFunc func(ctx context.Context, dev *ttnpb.EndDevice) types.DevAdd
 
 // makeNewDevAddrFunc returns a newDevAddrFunc, which derives DevAddrs using specified prefixes.
 func makeNewDevAddrFunc(ps ...types.DevAddrPrefix) newDevAddrFunc {
+	weights := make([]float64, len(ps))
+	for i, p := range ps {
+		weights[i] = math.Pow(2.0, float64(32-p.Length))
+	}
+	w := sampleuv.NewWeighted(weights, nil)
 	return func(ctx context.Context, dev *ttnpb.EndDevice) types.DevAddr {
 		var devAddr types.DevAddr
-		rand.Read(devAddr[:])
-		p := ps[random.Int63n(int64(len(ps)))]
-		return devAddr.WithPrefix(p)
+		_, _ = rand.Read(devAddr[:])
+		idx, _ := w.Take()
+		return devAddr.WithPrefix(ps[idx])
 	}
```

Comment:
This is missing tests, we don't need floating point operations (one bitshift will do) and I don't think we need a dependency for this.

We can just:

1. Loop over `ps` and put them in a new slice with a field `weight` that is `1 << (32-Length)`, and keep a `totalWeight`
2. Pick a random number `r` between 0 and `totalWeight` (inclusive)
3. Loop over the slice, decrease `r` by the `weight` in the slice, and when it goes `0` or lower, take that `ps`